File: Bai_68/main.py
def find_longest_increasing_subsequence(arr):
    n = len(arr)
    start = 0
    max_length = 1
    current_length = 1

    for i in range(1, n):
        if arr[i] > arr[i - 1]:
            current_length += 1
        else:
            if current_length > max_length:
                max_length = current_length
                start = i - max_length
            current_length = 1

    if current_length > max_length:
        start = n - current_length

    return arr[start : start + max_length]

a = [5, 2, 8, 6, 3, 6, 9, 7]
result = find_longest_increasing_subsequence(a)

print("Longest Increasing Subsequence: ", end="")
for num in result:
    print(num, end=" ")

File: Bai_68/main.cpp
#include <iostream>
#include <vector>

std::vector<int> findLongestIncreasingSubsequence(const std::vector<int>& arr) {
    int n = arr.size();
    int start = 0;
    int maxLength = 1;
    int currentLength = 1;

    for (int i = 1; i < n; ++i)
    {
        if (arr[i] > arr[i - 1])
            currentLength++;
        else
        {
            if (currentLength > maxLength)
            {
                maxLength = currentLength;
                start = i - maxLength;
            }
            currentLength = 1;
        }
    }

    if (currentLength > maxLength) 
        start = n - currentLength;

    return std::vector<int>(arr.begin() + start, arr.begin() + start + maxLength);
}

int main() {
    std::vector<int> a = {5, 2, 8, 6, 3, 6, 9, 7};

    std::vector<int> result = findLongestIncreasingSubsequence(a);

    std::cout << "Longest Increasing Subsequence: ";
    for (int num : result)
        std::cout << num << " ";
    return 0;
}